unity-client/Assets/ABConverter/VisualTests.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using DCL.Helpers;
+using System;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+using UnityEngine.Networking;
+using Object = UnityEngine.Object;
+using Random = UnityEngine.Random;
+
+namespace DCL.ABConverter
+{
+    public static class VisualTests
+    {
+        static readonly string abPath = Application.dataPath + "/../AssetBundles/";
+        static readonly string baselinePath = VisualTestHelpers.baselineImagesPath;
+        static readonly string testImagesPath = VisualTestHelpers.testImagesPath;
+
+        public static IEnumerator TestConvertedAssets(Core core = null, Environment env = null, Action<Core.ErrorCodes> OnFinish = null)
+        {
+            EditorSceneManager.OpenScene($"Assets/ABConverter/VisualTestScene.unity", OpenSceneMode.Single);
+
+            VisualTestHelpers.baselineImagesPath += "ABConverter/";
+            VisualTestHelpers.testImagesPath += "ABConverter/";
+
+            var gltfs = LoadAndInstantiateAllGltfAssets();
+
+            VisualTestHelpers.generateBaseline = true;
+
+            foreach (GameObject go in gltfs)
+            {
+                go.SetActive(false);
+            }
+
+            foreach (GameObject go in gltfs)
+            {
+                go.SetActive(true);
+                yield return VisualTestHelpers.TakeSnapshot($"ABConverter_{go.name}.png", Camera.main, new Vector3(7, 7, 7), Vector3.zero);
+                go.SetActive(false);
+            }
+
+            VisualTestHelpers.generateBaseline = false;
+
+            var abs = LoadAndInstantiateAllAssetBundles();
+
+            foreach (GameObject go in abs)
+            {
+                go.SetActive(false);
+            }
+
+            foreach (GameObject go in abs)
+            {
+                go.SetActive(true);
+                string testName = $"ABConverter_{go.name}.png";
+                yield return VisualTestHelpers.TakeSnapshot(testName, Camera.main, new Vector3(7, 7, 7), Vector3.zero);
+
+                bool result = VisualTestHelpers.TestSnapshot(
+                        VisualTestHelpers.baselineImagesPath + testName,
+                        VisualTestHelpers.testImagesPath + testName,
+                        95,
+                        false);
+
+                // Delete failed AB files to avoid uploading them
+                if (!result && env != null)
+                {
+                    string filePath = abPath + go.name;
+                    if (env.file.Exists(filePath))
+                    {
+                        env.file.Delete(filePath);
+
+                        string depMapPath = filePath + ".depmap";
+                        env.file.Delete(depMapPath);
+                    }
+
+                    if (core != null)
```

Comment:
The need to access the whole `Core` object just for error passing is a bit overkill. I think we can avoid this pretty easily.

I propose the following:
- Make `TestConvertedAssets` have its own `System.Action<int>` callback that pass along the number of skipped assets.
- From `Core` we can do something like:

```
EditorCoroutineUtility.StartCoroutineOwnerless(
        VisualTests.TestConvertedAssets(
            env: env, 
            OnFinish: (skippedAssets) => 
                {
                    this.skippedAssets = skippedAssets;
                    OnFinish?.Invoke(state.lastErrorCode);
                })
    );
```

Reply:
Well... Just figured this looks like I'm leaving review comments in my own code.

Reply:
Great suggestion, it bothered me as well that I needed the Core object to mark the "skipped assets".

Just implemented the change.